### DISCUSSION TITLE
生产者先于消费者启动，并在消费者启动之前投递了一些消息，消费者启动后为啥不能消费那些已投递但未消费的消息？

### DISCUSSION THREAD

**Discussion by zuoHx580** [apache/rocketmq#6702] [Q&A]
这种情况要怎么处理

**Comment by superhx** [comment]
1. Consumer 默认是 CONSUME_FROM_LAST_OFFSET，因此在没有消费进度的时候会优先从末尾消费（RocketMQ 优化场景：若队列 minOffset 为0，且数据还在内存中，则会从 0 开始消费）。
2. 如果想从头开始消费，可以设置为 CONSUME_FROM_FIRST_OFFSET

**Reply by n1nja88888** [reply]
> 若队列 minOffset 为0，且数据还在内存中，则会从 0 开始消费

想问一下这里这样设置的目的是什么？

**Comment by fuyou001** [comment]
此问题解决办法：重置消息位点，重新开始消费。

后续规避办法：建议先启动订阅方。
